File: Ruby/BasicAlgorithms/SquareArrayValues.rb
# Write a function that takes an integer array "numbers", and then multiplies each value by itself.
# For example, [1,5,10,-10] should become [1,25,100,100]
module SquareArrayValues

  def SquareArrayValues.For(numArray)
    squareArray = []
    for num in numArray
      squareArray.push(num * num)
    end
    print "For #{numArray} Square Array #{squareArray} \n" 
    return squareArray
  end

  def SquareArrayValues.While(numArray)
    iterator = 0
    squareArray = []
    begin
      square = numArray[iterator] * numArray[iterator] 
      squareArray.push (square)
      iterator += 1    
    end while iterator < numArray.length
    print "While #{numArray} Square Array #{squareArray} \n" 
    return squareArray
  end

  def SquareArrayValues.DoUntil(numArray)
    iterator = 0
    squareArray = []
    until iterator == numArray.length do
      square = numArray[iterator] * numArray[iterator] 
      squareArray.push (square) 
      iterator += 1      
    end
    print "Until #{numArray} Square Array #{squareArray} \n" 
    return squareArray
  end

end

File: Python/BasicAlgorithms/SquareArrayValues.py
# Write a function that takes an integer array "numbers", and then multiplies each value by itself.
# For example, [1,5,10,-10] should become [1,25,100,100]

def SquareArrayValues_For(numberArray):
    newArray = []
    for number in numberArray:
        newArray.append(number*number)
    print  (f"For {numberArray} Square Array {newArray}")

def SquareArrayValues_While(numberArray):
    iterator = 0
    newArray = []
    while iterator < len(numberArray):
        newArray.append(numberArray[iterator]*numberArray[iterator])
        iterator += 1
    print  (f"While {numberArray} Square Array {newArray}")